lib/vault/api/auth_token.rb

```diff
@@ -24,7 +24,12 @@ class AuthToken < Request
     #
     # @return [Secret]
     def create(options = {})
-      json = client.post("/v1/auth/token/create", JSON.fast_generate(options))
+      url = if options.include?(:role)
+              "/v1/auth/token/create/" + options.delete(:role)
+            else
+              "/v1/auth/token/create"
+            end
+      json = client.post(url, JSON.fast_generate(options))
```

Comment:
The delete method removes the key from the hash and returns the value, so it is no longer present in the options hash.

Verification:
```ruby
irb(main):001:0> h = {test: 'ok', role: 'myrole'}
=> {:test=>"ok", :role=>"myrole"}
irb(main):002:0> h.delete(:role)
=> "myrole"
irb(main):003:0> h
=> {:test=>"ok"}
```